src/decoder/plugins/SidplayDecoderPlugin.cxx

```diff
@@ -437,51 +439,83 @@ sidplay_file_decode(DecoderClient &client, Path path_fs)
 	} while (cmd != DecoderCommand::STOP);
 }
 
+static AllocatedString<char>
+Windows1252ToUTF8(const char *s)
+{
+#ifdef HAVE_ICU_CONVERTER
+	try {
+		std::unique_ptr<IcuConverter>
+			converter(IcuConverter::Create("windows-1252"));
+
+		return converter->ToUTF8(s);
+	} catch (...) { }
+#endif
+
+	/*
+	 * Fallback to not transcoding windows-1252 to utf-8, that may result
+	 * in invalid utf-8 unless nonprintable characters are replaced.
+	 */
+	AllocatedString<char> t = AllocatedString<char>::Duplicate(s);
+
+	for (size_t i = 0; t[i] != AllocatedString<char>::SENTINEL; i++)
+		if (!std::isprint(t[i]))
+			t[i] = '?';
+
+	return t;
+}
+
 gcc_pure
-static const char *
-GetInfoString(const SidTuneInfo &info, unsigned i) noexcept
+static AllocatedString<char>
+GetInfoString(const SidTuneInfo &info, unsigned i)
 {
 #ifdef HAVE_SIDPLAYFP
-	return info.numberOfInfoStrings() > i
+	const char *s = info.numberOfInfoStrings() > i
 		? info.infoString(i)
-		: nullptr;
+		: "";
 #else
-	return info.numberOfInfoStrings > i
+	const char *s = info.numberOfInfoStrings > i
 		? info.infoString[i]
-		: nullptr;
+		: "";
 #endif
+
+	return Windows1252ToUTF8(s);
 }
 
 static void
 ScanSidTuneInfo(const SidTuneInfo &info, unsigned track, unsigned n_tracks,
 		TagHandler &handler) noexcept
 {
-	/* title */
-	const char *title = GetInfoString(info, 0);
-	if (title == nullptr)
-		title = "";
-
-	if (n_tracks > 1) {
-		const auto tag_title =
-			StringFormat<1024>("%s (%u/%u)",
-					   title, track, n_tracks);
-		handler.OnTag(TAG_TITLE, tag_title.c_str());
-	} else
-		handler.OnTag(TAG_TITLE, title);
-
-	/* artist */
-	const char *artist = GetInfoString(info, 1);
-	if (artist != nullptr)
-		handler.OnTag(TAG_ARTIST, artist);
+	try {
+		/* title */
+		AllocatedString<char> title = GetInfoString(info, 0);
+
+		if (n_tracks > 1) {
+			const auto tag_title =
+				StringFormat<1024>("%s (%u/%u)",
+						   title.c_str(),
+						   track, n_tracks);
+			handler.OnTag(TAG_TITLE, tag_title.c_str());
+		} else
+			handler.OnTag(TAG_TITLE, title.c_str());
+	} catch (...) { }
```

Comment:
Exceptions have already been caught inside `Windows1252ToUTF8()`, so what is this supposed to catch?

Reply:
It seems AllocatedString may fail to allocate its string, and hence throw a memory related exception.

Reply:
As a general rule, I don't care for memory allocation failures; even if those would really throw an exception (they don't, at least not on Linux), it would be OK for me to abort MPD (via `std::terminate`). Once you're on memory shortage, there's little you can do to recover from the situation from inside the process; the best you can do to let the computer breathe again is just exit the process as quickly as possible. And MPD isn't a particularly important process.

Reply:
Ah, I was unaware that memory exceptions were exempted. I can remove those catches then.

Reply:
That was a good chance to improve the developer manual... https://github.com/MusicPlayerDaemon/MPD/commit/349a2ea7eb4df3d24bebe9061d47da37bfc078f4